todolist/src/redux/todoSlice.js

```diff
@@ -0,0 +1,95 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { v4 as uuidv4 } from "uuid";
+
+const initialState = {
+  todos: [],
+  currentTodo: null,
+};
+const todoSlice = createSlice({
+  name: "todo",
+  initialState,
+  reducers: {
+    addTodo: (state, action) => {
+      const todo = {
+        name: action.payload,
+        id: uuidv4(),
+        done: false,
+        createdAt: (new Date()).toISOString(),
+        important: false
+      };
+      state.todos.push(todo);
+    },
+    deleteTodo: (state, action) => {
+      const findIndexTodo = state.todos.findIndex(
+        (todo) => todo.id === action.payload
+      );
+      if (findIndexTodo !== -1) {
+        state.todos.splice(findIndexTodo, 1);
+      }
+    },
+    startEditTodo: (state, action) => {
+      const findTodo = state.todos.find((todo) => todo.id === action.payload);
+      if (findTodo) {
+        state.currentTodo = findTodo;
+      }
+    },
+    setCurrentTodo: (state, action) => {
+      state.currentTodo = action.payload;
+    },
+    setImportantTodo: (state, action) => {
+      const todoId = action.payload;
+      state.todos.some((todo, index) => {
+        if(todo.id === todoId) {
+          state.todos[index].important = !state.todos[index].important;
+          return true;
+        }
+        return false;
+      })
+    },
```

Comment:
Đọc lại bản chất hàm some, nó không dùng để lặp qua các phần tử như này, e có thể dùng for để lặp.
Người lại với hàm some là hàm gì e?

Reply:
Em dùng hàm some ở đây là để hạn chế nó lặp hết mảng ấy anh. Nếu có 1 phần tử nào đó thỏa mãn điều kiện thì nó k lặp tiếp ấy anh